compiler/luci-interpreter/src/kernels/FullyConnected.cpp

```diff
@@ -188,5 +206,30 @@ void FullyConnected::evalQuantizedS8() const
     getTensorShape(output()), getTensorData<int8_t>(output()));
 }
 
+void FullyConnected::evalHybridWI4AF32() const
+{
+  float activation_min{};
+  float activation_max{};
+  calculateActivationRange(_params.activation, &activation_min, &activation_max);
+
+  tflite::FullyConnectedParams params{};
+  params.float_activation_min = activation_min;
+  params.float_activation_max = activation_max;
+  params.weights_format = tflite::FullyConnectedWeightsFormat::kDefault;
+
+  const int8_t *weights_int4 = getTensorData<int8_t>(weights());
+  float *weights_float = getTensorData<float>(scratch());
+  const Shape &weights_shape = weights()->shape();
+  for (int32_t i = 0; i < weights_shape.num_elements(); ++i)
+  {
+    // 1bit for sign, 3bit for value
+    weights_float[i] = weights()->scale() * weights_int4[i];
+  }
+  tflite::reference_ops::FullyConnected(
```

Comment:
TFLite performs on-the-fly quantization to the input, but this kernel is based on fake quantization.

I think it is okay to go as-is because this kernel will be used mainly for testing (performance is not that important).